pkg/asset/installconfig/azure/validation.go

```diff
@@ -628,12 +629,22 @@ func ValidateForProvisioning(client API, ic *types.InstallConfig) error {
 	allErrs = append(allErrs, validateResourceGroup(client, field.NewPath("platform").Child("azure"), ic.Azure)...)
 	allErrs = append(allErrs, ValidateDiskEncryptionSet(client, ic)...)
 	allErrs = append(allErrs, ValidateSecurityProfileDiskEncryptionSet(client, ic)...)
+	allErrs = append(allErrs, validateSkipImageUpload(field.NewPath("image"), ic.ControlPlane.Platform)...)
```

Comment:
There should be a validation for compute nodes as well as the default machine platform.